tests/ignite/metrics/test_epoch_metric.py

```diff
@@ -51,73 +51,98 @@ def compute_fn(y_preds, y_targets):
         em.compute()
 
 
-def test_epoch_metric():
+def test_epoch_metric(available_device):
     def compute_fn(y_preds, y_targets):
         return 0.0
 
-    em = EpochMetric(compute_fn)
+    device = torch.device(available_device)
+    em = EpochMetric(compute_fn, device=device)
+    assert em._device == device
 
     em.reset()
-    output1 = (torch.rand(4, 3), torch.randint(0, 2, size=(4, 3), dtype=torch.long))
+    output1 = (torch.rand(4, 3, device=device), torch.randint(0, 2, size=(4, 3), dtype=torch.long, device=device))
```

Comment:
You set both the metric input and metric internal accumulators to available device (i.e. `EpochMetric(compute_fn, device=device)`), but we first wanted to test input on cpu and metric's accumulator on available device.

Reply:
do I understand it correctly that output1 and output2 are inputs and that in the first part of the test they should be on cpu?

Reply:
output1 and output2 should always be on cpu. yes, they are inputs to the metric and outputs from the model + ground truth.

Reply:
I don't understand how you would compare tensors on different devices :(